airgun/entities/subscription.py

```diff
@@ -195,17 +203,32 @@ class SubscriptionDetails(NavigateStep):
 
     Args:
         entity_name: name of Subscription
+        virt_who: Whether this is a virt who client subscription.
     """
     VIEW = SubscriptionDetailsView
 
     def prerequisite(self, *args, **kwargs):
         return self.navigate_to(self.obj, 'All')
 
     def am_i_here(self, *args, **kwargs):
-        subscription_name = kwargs.get('subscription')
+        subscription_name = kwargs.get('entity_name')
         return (self.view.is_displayed and
                 self.view.breadcrumb.read() == subscription_name)
 
     def step(self, *args, **kwargs):
         subscription_name = kwargs.get('entity_name')
+        virt_who = kwargs.get('virt_who')
+        search_string = 'name = "{0}"'.format(subscription_name)
+        if virt_who:
+            search_string = 'virt_who = true and {0}'.format(search_string)
+        self.parent.search(search_string)
         self.parent.table.row(name=subscription_name)['Name'].widget.click()
+        wait_for_spinner(self.view)
+        self.parent.browser.plugin.ensure_page_safe()
+        wait_for(
```

Comment:
+1 here
we have too many methods that validate our navigation is done. Can we have something more generic?